pkg/storage/storagebase/proposer_kv.proto

```diff
@@ -172,7 +172,7 @@ message RaftCommand {
 
   // Proposer-evaluated KV mode.
 
-  optional ReplicatedEvalResult replicated_eval_result = 13;
+  optional ReplicatedEvalResult replicated_eval_result = 13 [(gogoproto.nullable) = false];
```

Comment:
Is there a migration concern here? That is, does the previous version knows how to handle empty `ReplicatedEvalResult`? Seeing that there are various `!= nil` checks here, can one of them bite us during a rolling upgrade in which an old version receives a `ReplicatedEvalResult{}` instead of `nil` as it expects?